File: app/src/main/java/com/prox/voicechanger/ui/fragment/SettingFragment.java
package com.prox.voicechanger.ui.fragment;

import static com.prox.voicechanger.VoiceChangerApp.TAG;

import android.content.Intent;
import android.graphics.Color;
import android.graphics.Shader;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;

import com.prox.voicechanger.R;
import com.prox.voicechanger.databinding.DialogRateBinding;
import com.prox.voicechanger.databinding.FragmentSettingBinding;
import com.prox.voicechanger.ui.dialog.RateDialog;
import com.prox.voicechanger.utils.ColorUtils;

public class SettingFragment extends Fragment {
    private static final String URI_POLICY = "https://hellowordapp.github.io/policy/privacy.html";

    private FragmentSettingBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        Log.d(TAG, "SettingFragment: onCreateView");
        binding = FragmentSettingBinding.inflate(inflater, container, false);

        init();

        binding.btnClose.setOnClickListener(view -> {
            NavController navController= Navigation.findNavController(requireActivity(), R.id.nav_host_record_activity);
            navController.popBackStack();
            Log.d(TAG, "SettingFragment: To RecordFragment");
        });

        binding.btnRate.setOnClickListener(view -> {
            RateDialog dialog = new RateDialog(
                    requireContext(),
                    DialogRateBinding.inflate(requireActivity().getLayoutInflater()));
            dialog.show();
            Log.d(TAG, "SettingFragment: Show RateDialog");
        });

        binding.btnShareApp.setOnClickListener(view -> {
            Intent intent = new Intent(Intent.ACTION_SEND);
            intent.setType("text/plain");
            intent.putExtra(Intent.EXTRA_SUBJECT, getResources().getString(R.string.app_name));
            intent.putExtra(Intent.EXTRA_TEXT, "https://play.google.com/store/apps/details?id="+requireActivity().getApplicationContext().getPackageName());
            startActivity(Intent.createChooser(intent, getResources().getString(R.string.share_to)));
        });

        binding.btnPolicy.setOnClickListener(view -> {
            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(URI_POLICY)));
            requireActivity().overridePendingTransition(R.anim.anim_right_left_1, R.anim.anim_right_left_2);
            Log.d(TAG, "SettingFragment: To "+URI_POLICY);
        });

        return binding.getRoot();
    }

    private void init() {
        Log.d(TAG, "SettingFragment: init");
        Shader shader1 = ColorUtils.textShader(
                Color.parseColor("#4B5DFC"),
                Color.parseColor("#F7277E"),
                binding.layoutPremium.txtPremium1.getTextSize());
        Shader shader2 = ColorUtils.textShader(
                Color.parseColor("#4B5DFC"),
                Color.parseColor("#F7277E"),
                binding.layoutPremium.txtPremium2.getTextSize());
        binding.layoutPremium.txtPremium1.getPaint().setShader(shader1);
        binding.layoutPremium.txtPremium2.getPaint().setShader(shader2);
    }
}

File: app/src/main/java/com/prox/voicechanger/ui/home/fragment/SettingFragment.kt
package com.prox.voicechanger.ui.home.fragment

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import com.prox.voicechanger.R
import android.content.Intent
import android.graphics.Color
import android.net.Uri
import android.view.View
import androidx.fragment.app.Fragment
import com.ntduc.contextutils.shareText
import com.ntduc.toastutils.shortToast
import com.prox.voicechanger.VoiceChangerApp
import com.prox.voicechanger.databinding.DialogRateBinding
import com.prox.voicechanger.databinding.FragmentSettingBinding
import com.prox.voicechanger.ui.dialog.RateDialog
import com.prox.voicechanger.utils.ColorUtils

class SettingFragment : Fragment() {
    companion object {
        private const val URI_POLICY = "https://hellowordapp.github.io/policy/privacy.html"
    }

    private lateinit var binding: FragmentSettingBinding
    private var callback: Callback? = null

    fun newInstance(): SettingFragment {
        val args = Bundle()
        val fragment = SettingFragment()
        fragment.arguments = args
        return fragment
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentSettingBinding.inflate(inflater)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        init()
    }

    private fun init() {
        initView()
        initEvent()
    }

    private fun initView() {
        val shader1 = ColorUtils.textShader(
            Color.parseColor("#4B5DFC"),
            Color.parseColor("#F7277E"),
            binding.layoutPremium.txtPremium1.textSize
        )
        val shader2 = ColorUtils.textShader(
            Color.parseColor("#4B5DFC"),
            Color.parseColor("#F7277E"),
            binding.layoutPremium.txtPremium2.textSize
        )
        binding.layoutPremium.txtPremium1.paint.shader = shader1
        binding.layoutPremium.txtPremium2.paint.shader = shader2
    }

    private fun initEvent() {
        binding.btnClose.setOnClickListener {
            callback?.onClose()
        }

        binding.btnRate.setOnClickListener {
            if (RateDialog.isRated(requireContext())) {
                requireContext().shortToast(R.string.rated)
            } else {
                val dialog = RateDialog(
                    requireContext(),
                    DialogRateBinding.inflate(requireActivity().layoutInflater)
                ) {}
                dialog.show()
            }
        }

        binding.btnShareApp.setOnClickListener {
            requireContext().shareText(
                subject = resources.getString(R.string.app_name),
                text = "https://play.google.com/store/apps/details?id=" + requireActivity().applicationContext.packageName
            )
        }

        binding.btnPolicy.setOnClickListener {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(URI_POLICY)))
            requireActivity().overridePendingTransition(
                R.anim.anim_right_left_1,
                R.anim.anim_right_left_2
            )
        }
    }

    fun setCallback(callback: Callback) {
        this.callback = callback
    }

    interface Callback {
        fun onClose()
    }
}